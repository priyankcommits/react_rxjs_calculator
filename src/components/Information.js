// Component to show key board shorts cuts
import Badge from 'react-bootstrap/Badge';
import Button from 'react-bootstrap/Button';
import Col from 'react-bootstrap/Col';
import Modal from 'react-bootstrap/Modal';
import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';

function Information(props) {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Button variant='dark' onClick={handleShow} size='sm' className='m-1 ml-3'>
        i
      </Button>

      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>
            Keyboard Shorcuts
            <Badge variant='secondary' className='ml-2'>
              <h6>
                Button : Shortcut
              </h6>
            </Badge>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Row>
              {props.buttons.map((button, index) => (
                <Col xs={4} className='my-1' key={index}>
                  <Badge variant='secondary'>
                    <h5>
                      {button.value.name} : {button.keyEvent}
                    </h5>
                  </Badge>
                </Col>
              ))}
          </Row>
        </Modal.Body>
      </Modal>
    </>
  );
}

export default Information;
