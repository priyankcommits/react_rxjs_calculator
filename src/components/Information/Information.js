// Component to show key board shorts cuts
import Badge from 'react-bootstrap/Badge';
import Button from 'react-bootstrap/Button';
import Col from 'react-bootstrap/Col';
import Modal from 'react-bootstrap/Modal';
import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';

import { RoundedButton } from '../../shared/styles/main/index';

const Information = (props) => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Button variant='dark' onClick={handleShow} size='sm' className='mt-1 ml-2' as={RoundedButton}>
        <span className='mx-1'>i</span>
      </Button>

      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>
            Keyboard Shorcuts
            <Badge pill variant='success' className='ml-2'>
              Button : Shortcut
            </Badge>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Row>
              {props.buttons.map((button, index) => (
                <Col xs={4} className='mt-1' key={index}>
                  <Badge pill variant='primary'>
                    {button.value.name}
                  </Badge>
                  <span className='mx-1'>:</span>
                  <Badge pill variant='dark'>
                    {button.keyEvent}
                  </Badge>
                </Col>
              ))}
          </Row>
        </Modal.Body>
      </Modal>
    </>
  );
}

export { Information };
