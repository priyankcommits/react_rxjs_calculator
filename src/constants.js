export const buttonTypes = {
    NUMBER: 'number',
    OPERATOR: 'operator',
    FUNCTION: 'function',
    RESET: 'reset',
}

export const operators = {
    PLUS: '+',
    MINUS: '-',
    MULTIPLY: 'x',
    DIVIDE: '÷',
}

const claculatorButtons = [
    {value: {name: 'sin', startsWith: 'sin(', endsWith: ')'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: 'ln', startsWith: 'ln(', endsWith: ')'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: '7'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '8'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '9'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: operators.DIVIDE}, type: buttonTypes.OPERATOR, color: 'secondary'},
    {value: {name: 'cos', startsWith: 'cos(', endsWith: ')'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: 'log', startsWith: 'log(', endsWith: ')'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: '4'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '5'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '6'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: operators.MULTIPLY}, type: buttonTypes.OPERATOR, color: 'secondary'},
    {value: {name: 'tan', startsWith: 'tan(', endsWith: ')'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: 'π', startsWith: '', endsWith: 'π'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: '1'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '2'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '3'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: operators.MINUS}, type: buttonTypes.OPERATOR, color: 'secondary'},
    {value: {name: 'pow', startsWith: '^', endsWith: ''}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: 'e', startsWith: '', endsWith: 'e'}, type: buttonTypes.FUNCTION, color: 'secondary'},
    {value: {name: '0'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: '.'}, type: buttonTypes.NUMBER, color: 'light'},
    {value: {name: 'AC'}, type: buttonTypes.RESET, color: 'primary'},
    {value: {name: operators.PLUS}, type: buttonTypes.OPERATOR, color: 'secondary'}
]

export default claculatorButtons;
