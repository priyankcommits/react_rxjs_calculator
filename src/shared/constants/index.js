// Contains all constants related to Appplication

const buttonTypes = {
  NUMBER: 'number',
  OPERATOR: 'operator',
  FUNCTION: 'function',
  CONSTANT: 'constant',
  RESET: 'AC',
  PARANTHESIS_OPEN: '(',
  PARANTHESIS_CLOSE: ')',
  POINT: '.',
  ANSWER: 'Ans',
  UNDO: 'Undo',
  REDO: 'Redo',
}

const operators = {
  PLUS: '+',
  MINUS: '-',
  MULTIPLY: 'x',
  DIVIDE: 'รท',
  POWER: 'pow',
  MODULUS: '%',
}

const calculatorButtons = [
  {value: {name: 'sin'}, type: buttonTypes.FUNCTION, color: 'secondary', keyEvent: 's'},
  {value: {name: 'ln'}, type: buttonTypes.FUNCTION, color: 'secondary', keyEvent: 'l'},
  {value: {name: '7'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '7'},
  {value: {name: '8'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '8'},
  {value: {name: '9'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '9'},
  {value: {name: operators.DIVIDE}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '/'},
  {value: {name: 'cos'}, type: buttonTypes.FUNCTION, color: 'secondary', keyEvent: 'c'},
  {value: {name: 'log'}, type: buttonTypes.FUNCTION, color: 'secondary', keyEvent: 'g'},
  {value: {name: '4'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '4'},
  {value: {name: '5'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '5'},
  {value: {name: '6'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '6'},
  {value: {name: operators.MULTIPLY}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '*'},
  {value: {name: 'tan'}, type: buttonTypes.FUNCTION, color: 'secondary', keyEvent: 't'},
  {value: {name: 'ฯ'}, type: buttonTypes.CONSTANT, color: 'secondary', keyEvent: 'p'},
  {value: {name: '1'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '1'},
  {value: {name: '2'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '2'},
  {value: {name: '3'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '3'},
  {value: {name: operators.MINUS}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '-'},
  {value: {name: operators.POWER}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '^'},
  {value: {name: 'e'}, type: buttonTypes.CONSTANT, color: 'secondary', keyEvent: 'e'},
  {value: {name: '0'}, type: buttonTypes.NUMBER, color: 'light', keyEvent: '0'},
  {value: {name: buttonTypes.POINT}, type: buttonTypes.POINT, color: 'light', keyEvent: '.'},
  {value: {name: buttonTypes.RESET}, type: buttonTypes.RESET, color: 'primary', keyEvent: 'r'},
  {value: {name: operators.PLUS}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '+'},
  {value: {name: buttonTypes.PARANTHESIS_OPEN}, type: buttonTypes.PARANTHESIS_OPEN, color: 'secondary', keyEvent: '('},
  {value: {name: buttonTypes.PARANTHESIS_CLOSE}, type: buttonTypes.PARANTHESIS_CLOSE, color: 'secondary', keyEvent: ')'},
  {value: {name: buttonTypes.UNDO}, type: buttonTypes.UNDO, color: 'warning', keyEvent: 'z'},
  {value: {name: buttonTypes.REDO}, type: buttonTypes.REDO, color: 'warning', keyEvent: 'x'},
  {value: {name: buttonTypes.ANSWER}, type: buttonTypes.CONSTANT, color: 'primary', keyEvent: 'a'},
  {value: {name: operators.MODULUS}, type: buttonTypes.OPERATOR, color: 'secondary', keyEvent: '%'},
]

export { buttonTypes, calculatorButtons, operators };
