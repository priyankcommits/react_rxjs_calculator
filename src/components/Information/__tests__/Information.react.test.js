// Snapshot testing
import React from 'react';
import renderer from 'react-test-renderer';

import { Information } from '../Information';
import { calculatorButtons } from '../../../shared/constants/index';

test('Information Initial Rendering', () => {
  const component = renderer.create(
    <Information buttons={calculatorButtons} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
