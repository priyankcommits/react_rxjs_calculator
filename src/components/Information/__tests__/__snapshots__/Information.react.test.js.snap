// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Initial Rendering 1`] = `
<div
  className="sc-AxhCb kgMfC mt-1 ml-2 btn btn-dark btn-sm"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="mx-1"
  >
    i
  </span>
</div>
`;
